ckanext/canada/controller.py

```diff
@@ -817,10 +817,18 @@ def create_pd_record(self, owner_org, resource_name):
         except NotAuthorized:
             return abort(403, _('Unauthorized'))
 
-        choice_fields = {
-            f['datastore_id']: [
-                {'value': k, 'label': v} for (k, v) in f['choices']]
-            for f in h.recombinant_choice_fields(resource_name)}
+        choice_fields = {}
+        for f in h.recombinant_choice_fields(resource_name):
+            for r in chromo['fields']:
+                if r['datastore_id'] == f['datastore_id'] and \
+                        r.get('form_full_text_choices', False):
+                    choice_fields[f['datastore_id']] = [{'value': k,
+                                                         'label': k + ': ' + v}
+                                                        for (k, v) in f['choices']]
+            if f['datastore_id'] not in choice_fields:
+                choice_fields[f['datastore_id']] = [{'value': k, 'label': v}
+                                                    for (k, v) in f['choices']]
+
```

Comment:
Let's take the loop that doesn't depend on `f` out front to avoid unnecessary O(n**2)

```suggestion
        for f in chromo['fields']:
            if f['datastore_id'] == r['datastore_id']:
                ft_choices = f.get('form_full_text_choices', False)
                break

        choice_fields = {
            f['datastore_id']: [
                {
                    'value': k, 
                    'label': k + ': ' + v if ft_choices else: v,
                } for (k, v) in f['choices']
            ] for f in h.recombinant_choice_fields(resource_name)
        }
```

Reply:
unfortunately, `if f['datastore_id'] == r['datastore_id']` won't work as I need to find the value of `form_full_text_choices` for each chromo field that I find in the `recombinant_choice_fields(resource_name)` so I have to double loop. However, I can optimize the code to improve time complexity.